code/game/gamemodes/umbrage/umbrage_unsorted.dm

```diff
@@ -0,0 +1,66 @@
+//The umbrage mutant species. They have no blood, don't need to breathe, are immune to radiation and viruses, and, most importantly, cannot wield guns. They also have night vision.
+/datum/species/umbrage
+	name = "Umbrage"
+	id = "umbrage"
+	darksight = 8
+	invis_sight = SEE_INVISIBLE_MINIMUM
+	sexes = 0
+	blacklisted = 1
+	say_mod = "chirps"
+	meat = /obj/item/weapon/reagent_containers/food/snacks/meat/slab/human/mutant/shadow
+	species_traits = list(NOBREATH, NOBLOOD, RADIMMUNE, VIRUSIMMUNE, NOGUNS)
+	dangerous_existence = 1
+	var/datum/action/innate/shadow/darkvision/vision_toggle
+
+/datum/species/umbrage/on_species_gain(mob/living/carbon/C, datum/species/old_species)
+	. = ..()
+	vision_toggle = new
+	vision_toggle.Grant(C)
+
+/datum/species/umbrage/on_species_loss(mob/living/carbon/C)
+	. = ..()
+	if(vision_toggle)
+		vision_toggle.Remove(C)
+
+/datum/species/umbrage/spec_life(mob/living/carbon/human/H)
+	var/turf/T = get_turf(H)
+	var/lumcount = T.get_lumcount()
+	if(H.loc == T)
+		if(lumcount > 3) //If it's light, we die very quickly
+			H << "<span class='userdanger'>The light burns you!</span>"
+			H << sound('sound/weapons/sear.ogg', volume = 75) //Spam text and a sound to aggressively say "hey, you're dying"
+			H.adjustFireLoss(UMBRAGE_LIGHT_BURN)
+		if(lumcount < 4) //But if it's dark, we heal, albeit slowly
+			H.adjustBruteLoss(UMBRAGE_DARK_HEAL)
```

Comment:
If you want, you could make this heal X damage in a specific order instead of healing all types at once.

    var/static/list/damage_heal_order = list(CLONE, BURN, BRUTE, TOX, OXY)
    H.heal_ordered_damage(whatever value here, damage_heal_order)